### PR TITLE
Add awesome tests

### PR DESCRIPTION
**Pull request by asdil12**
see https://progress.opensuse.org/issues/9522

### CODE REVIEW

File: tests/x11/awesome_menu.pm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+use base "x11test";
+use testapi;
+
+sub run() {
+    my $self = shift;
+    send_key "super-w";
+	sleep 1;
+    mouse_hide(1);
+	sleep 1;
```
Comment (okurz) [inline tests/x11/awesome_menu.pm]: travis checks fail here. You can (and should) execute tests locally before submitting. Call
`make test`
in your working copy to call the tests.

File: tests/console/consoletest_finish.pm [hunk 2]
```diff
@@ -58,7 +58,12 @@ sub run() {
         }
         wait_idle;
         mouse_hide(1);
-        assert_screen 'generic-desktop';
+        if (check_var("DESKTOP", "awesome")) {
+            assert_screen 'displaymanager', 200;
```
Comment (okurz) [inline tests/console/consoletest_finish.pm]: Can you explain why `assert_screen 'generic-desktop';` does not work for awesome tests? the needles look the same except for the tags, maybe?

File: tests/console/install_awesome.pm [hunk 3]
```diff
@@ -0,0 +1,19 @@
+use base "consoletest";
+use testapi;
+
+sub run() {
+    my $self = shift;
+    become_root();
+    #assert_script_run("zypper -n in awesome awesome-branding-openSUSE awesome-freedesktop awesome-shifty awesome-vicious");
```
Comment (okurz) [inline tests/console/install_awesome.pm]: don't commit disabled code

File: tests/console/install_awesome.pm [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    assert_script_run("zypper -n in awesome");
+	script_run("echo -e '#!/bin/sh\\n\\nexec dbus-launch --exit-with-session awesome' > /home/$username/.xinitrc");
+	script_run("cat /home/$username/.xinitrc");
```
Comment (okurz) [inline tests/console/install_awesome.pm]: that shouldn't be necessary.

File: tests/x11/awesome_xterm.pm [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+use testapi;
+
+sub run() {
+    my $self = shift;
+    mouse_hide(1);
+    send_key "super-r";
+    type_string "xterm\n";
+    sleep 2;
+    type_string "cd\n";
+    sleep 1;              # go to $HOME (for KDE)
```
Comment (okurz) [inline tests/x11/awesome_xterm.pm]: why the sleeps? what is the relation to KDE?

File: tests/x11/awesome_xterm.pm [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    send_key "ctrl-l";    # clear
+    for (1 .. 13) { send_key "ret" }
```
Comment (okurz) [inline tests/x11/awesome_xterm.pm]: why 13?

**Comment by asdil12** [comment]
Ran through completely: http://artemis/tests/536
Started using this command:
```
/var/lib/openqa/script/client isos post iso=openSUSE-Tumbleweed-DVD-x86_64-Snapshot20160105-Media.iso DISTRI=opensuse VERSION=Tumbleweed FLAVOR=DVD ARCH=x86_64 TEST=awesome MAKETESTSNAPSHOTS=1 KEEPHDDS=1
```

File: .gitignore [hunk 7]
```diff
@@ -1 +1,2 @@
 needles
+.*.swp
```
Comment (okurz) [inline .gitignore]: Unrelated change

File: main.pm [hunk 8]
```diff
@@ -549,6 +553,12 @@ sub load_x11tests() {
     if (get_var("NOAUTOLOGIN") || get_var("XDMUSED")) {
         loadtest "x11/x11_login.pm";
     }
+    if (check_var("DESKTOP", "awesome")) {
```
Comment (okurz) [inline main.pm]: Please add a comment why you do this

File: tests/installation/first_boot.pm [hunk 9]
```diff
@@ -10,6 +10,11 @@ sub run() {
         return;
     }
 
+    if (check_var('DESKTOP', 'awesome')) {
```
Comment (okurz) [inline tests/installation/first_boot.pm]: Please add a comment why you do this

Comment (nilxam) [inline tests/installation/first_boot.pm]: well, that check_var('DESKTOP', 'awesome') should be intentional to go away of the test since it's NOAUTOLOGIN test :)

File: tests/console/consoletest_finish.pm (hunk 2 above)
Comment (nilxam) [inline tests/console/consoletest_finish.pm]: hmm...interesting part, this test I assumed it will brings x11_login test, so in this stage it must stop here and skip logging in but log in to awesome in next step ie. x11_login. I didn't say it's strange or wrong, it's just a rare case use x11_login :)

File: tests/installation/first_boot.pm (hunk 9 above)
Comment (okurz) [inline tests/installation/first_boot.pm]: @nilxam thanks for your comments. Maybe I can update the commits with the comments and merge so that we don't have to wait and @asdil12 does not have to do another iteration.

File: tests/installation/installer_desktopselection.pm [hunk 10]
```diff
@@ -6,7 +6,7 @@ use testapi;
 sub run() {
     my %desktopkeys = (kde => "k", gnome => "g", xfce => "x", lxde => "l", minimalx => "m", textmode => "i");
     assert_screen "desktop-selection", 30;
-    my $d   = get_var("DESKTOP");
+    my $d = get_var("DESKTOP") != "awesome" ? get_var("DESKTOP") : "minimalx";
```
Comment (coolo) [inline tests/installation/installer_desktopselection.pm]: that's so not perl

File: tests/x11/awesome_xterm.pm [hunk 11]
```diff
@@ -0,0 +1,19 @@
+use base "x11test";
+use testapi;
+
+sub run() {
+    my $self = shift;
+    mouse_hide(1);
```
Comment (coolo) [inline tests/x11/awesome_xterm.pm]: why the mouse_hide here? the mouse should be pretty much hidden in every test before.

File: tests/installation/installer_desktopselection.pm (hunk 10 above)
Comment (okurz) [inline tests/installation/installer_desktopselection.pm]: what do *you* recommend?

Comment (coolo) [inline tests/installation/installer_desktopselection.pm]: comparing perl strings using perl string operators

Comment (aaannz) [inline tests/installation/installer_desktopselection.pm]: And what about `!check_var('DESKTOP', 'awesome') ? get_var("DESKTOP") : "minimalx";`

Comment (okurz) [inline tests/installation/installer_desktopselection.pm]: makes sense, thx

**Comment by okurz** [comment]
Obsoleted by #930

All comments taken into account and simplified test
